src/components/ContactsApp.jsx

```diff
@@ -1,47 +1,75 @@
 import React from "react";
 import ContactsList from "./ContactsList/ContactsList";
-import { withContacts } from "./ContactsList";
 import ContactsHeader from "./ContactsList/ContactsHeader";
 import ContactCard from "./ContactCard/ContactCard";
 import Modal, { ModalContent } from "cozy-ui/react/Modal";
+import { Button } from "cozy-ui/react/Button";
 import { translate } from "cozy-ui/react/I18n";
-
-const ConnectedContactsList = withContacts(ContactsList);
+import { PropTypes } from "prop-types";
 
 const TranslatedContactCard = translate()(({ t, ...props }) => (
   <ContactCard title={t("contact_info")} {...props} />
 ));
 
+const ContactActions = ({ contact, deleteContact }) => (
+  <div>
+    <Button theme="danger" onClick={() => deleteContact(contact)}>
+      delete
+    </Button>
+  </div>
+);
+
 class ContactsApp extends React.Component {
+  static propTypes = {
+    contacts: PropTypes.array,
+    deleteContact: PropTypes.func
+  };
+
   state = {
     displayedContact: null
   };
 
-  showContact = contact => {
+  displayContactCard = contact => {
     this.setState({
       displayedContact: contact
     });
   };
 
-  hideContact = () => {
+  hideContactCard = () => {
     this.setState({
       displayedContact: null
     });
   };
 
+  onDeleteContact = contact => {
+    this.props.deleteContact(contact);
+    this.hideContactCard();
+  };
```

Comment:
The component `<ContactsApp />` is reponsible (at least for now) to open and close the modale.
Maybe I can condition the `hideContactCard` action with the result of the `deleteContact` action.